### PR TITLE
Feat: require approval deployment + approve/cancel commands

### PR DESCRIPTION
**Pull request by yaronya**
### Feature Request
1. A user should be able to deploy/destroy an environment with some requires-approval flag.
2. A user should be able to approve an awaiting deployment from CLI.
3. A user should be able to cancel an awaiting deployment from CLI.

### Solution
1. New `--requireApproval` (`-a`) argument.
2. New `env0 approve` command.
3. New `env0 cancel` command.

### CODE REVIEW

**Comment by yaronya** [comment]
[CLI Plan](https://trello.com/c/um0OwD7F/1650-cli-plan)

File: node/tests/env0-deploy-utils.spec.js [hunk 1]
```diff
@@ -207,4 +207,35 @@ describe("env0-deploy-utils", () => {
...
+    const mockBlueprintId = 'blueprint0';
+    const mockRequiresApproval = true;
+
+    await deployUtils.deployEnvironment(mockEnvironment, mockBlueprintRevision, mockBlueprintId, mockRequiresApproval);
+
+    expect(mockCallApi).toHaveBeenLastCalledWith(
+        'post',
+        `environments/${mockEnvironment.id}/deployments`,
+        { data: { blueprintId: mockBlueprintId, blueprintRevision: mockBlueprintRevision, userRequiresApproval: mockRequiresApproval } }
+        )
```
Comment (RLRabinowitz) [inline node/tests/env0-deploy-utils.spec.js:223]: There is no `it` wrapping this

File: node/src/env0-deploy-utils.js [hunk 2]
```diff
@@ -136,6 +150,10 @@ class DeployUtils {
     do {
       ({ status } = await this.apiClient.callApi('get', `environments/${environmentId}`));
 
+      if (status === 'WAITING_FOR_USER') {
+        throw new Error('Deployment is waiting for an approval. Run \'env0 approve\' or \'env0 cancel\' to continue.');
+      }
+
       if (environmentValidStatuses.includes(status)) return;
```
Comment (RLRabinowitz) [inline node/src/env0-deploy-utils.js:157]: Add CANCELED to `environmentValidStatuses`

Comment (RLRabinowitz) [reply node/src/env0-deploy-utils.js:157]: Also, I have already merged the new ABORTED status, so I recommend adding it as well

File: node/src/env0-deploy-cli.js [hunk 3]
```diff
@@ -18,32 +19,25 @@ const {
...
+    options: allArguments,
     description: 'Deploys an environment'
   },
   'destroy': {
-    options: optionDefinitions,
+    options: allArguments,
     description: 'Destroys an environment'
   },
+  'approve': {
+    options: [ ...baseArguments, argumentsMap[ENVIRONMENT_NAME] ],
```
Comment (RLRabinowitz) [inline node/src/env0-deploy-cli.js:34]: Why isn't `ENVIRONMENT_NAME` part of `baseArguments`? It seems to be available in every command

File: node/src/env0-deploy-cli.js [hunk 4]
```diff
@@ -18,32 +19,25 @@ const {
...
-  { name: ENVIRONMENT_VARIABLES, alias: 'v', type: String, multiple: true, defaultValue: [], description: 'The environment variables to set on the deployed environment - works only on deploy and can be multiple, the format is "environmentVariableName1=value"' },
-  { name: SENSITIVE_ENVIRONMENT_VARIABLES, alias: 'q', type: String, multiple: true, defaultValue: [], description: 'The sensitive environment variables to set on the deployed environment - works only on deploy and can be multiple, the format is "environmentVariableName1=value"' },
-  { name: REVISION, alias: 'r', type: String, defaultValue: 'master', description: 'Your git revision, can be a branch tag or a commit hash. Default value "master" ' },
-  { name: ARCHIVE_AFTER_DESTROY, type: Boolean, defaultValue: false, description: 'Archive the environment after a successful destroy' },
-];
-
 const availableCommands = {
   'deploy': {
-    options: optionDefinitions,
+    options: allArguments,
```
Comment (RLRabinowitz) [inline node/src/env0-deploy-cli.js:26]: Does this work? This used to be an object and now it's an array

Comment (yaronya) [reply node/src/env0-deploy-cli.js:26]: It was an array even before 🤭

File: node/tests/env0-deploy-utils.spec.js (hunk 1 above)
Comment (yaronya) [reply node/tests/env0-deploy-utils.spec.js]: 🤦 
ace0734

File: node/src/env0-deploy-utils.js (hunk 2 above)
Comment (yaronya) [reply node/src/env0-deploy-utils.js:241]: e721896 e291656

File: node/src/env0-deploy-cli.js (hunk 3 above)
Comment (yaronya) [reply node/src/env0-deploy-cli.js]: bf182e0

File: node/src/commons/arguments.js [hunk 5]
```diff
@@ -31,5 +31,5 @@ const argumentsMap = {
 module.exports = {
     argumentsMap,
```
Comment (RLRabinowitz) [inline node/src/commons/arguments.js:32]: I think `argumentsMap` no longer needs to be exported

Comment (yaronya) [reply node/src/commons/arguments.js]: indeed e82a080

**Pull request merged**